traits/tests/test_interfaces.py

```diff
@@ -203,6 +236,47 @@ def test_provides_bad(self):
             class Test(HasTraits):
                 pass
 
+    def test_provides_with_no_interface_check(self):
+
+        class Test(HasTraits):
+            # Deliberately _not_ implementing get_foo. This class
+            # should not pass an IFoo interface check.
+            pass
+
+        provides_ifoo = provides(IFoo)
+        with self.set_check_interfaces(0):
+            # Simulate application of the decorator
+            Test = provides_ifoo(Test)
+
+        test = Test()
+        self.assertIsInstance(test, IFoo)
+
+    def test_provides_with_interface_check_warn(self):
+
+        class Test(HasTraits):
+            # Deliberately _not_ implementing get_foo. This class
+            # should not pass an IFoo interface check.
+            pass
+
+        provides_ifoo = provides(IFoo)
+        with self.set_check_interfaces(1):
+            with self.assertLogs("traits", logging.WARNING):
+                # Simulate application of the decorator
+                Test = provides_ifoo(Test)
+
+    def test_provides_with_interface_check_error(self):
+
+        class Test(HasTraits):
+            # Deliberately _not_ implementing get_foo. This class
+            # should not pass an IFoo interface check.
+            pass
+
+        provides_ifoo = provides(IFoo)
+        with self.set_check_interfaces(2):
+            with self.assertRaises(InterfaceError):
+                # Simulate application of the decorator
+                Test = provides_ifoo(Test)
```

Comment:
Is the assignment `Test = ` necessary here?

Reply:
It's not. I'm surprised I didn't get a `flake8` warning. I'll remove it, and the similar occurrence in `test_provides_with_interface_check_warn`.

Reply:
Though actually, it may be worth adding explicit tests that in the warning case, we _do_ end up with something that implements `IFoo`, and in the error case, we don't.

Reply:
Either way sounds good to me!

Reply:
Hmm. It turns out that the extra test I added failed: even though the application of `provides_ifoo` fails in this case, `Test` _still_ gets registered as implementing `IFoo`! It doesn't seem worth fixing, for something we're deprecating anyway, but now I'm not sure whether to encode the behaviour in a test or just leave it out.

On balance, I think I'll keep the extra test, but I could be persuaded otherwise.

Reply:
Extra tests added; unused variable is now used!

Reply:
> Hmm. It turns out that the extra test I added failed: even though the application of provides_ifoo fails in this case, Test still gets registered as implementing IFoo! It doesn't seem worth fixing, for something we're deprecating anyway, but now I'm not sure whether to encode the behaviour in a test or just leave it out.

Yep this is surprising, and I agree that this is not worth fixing.